### ISSUE TITLE
pintk is not respecting JUMPs defined in .tim files

### ISSUE THREAD

**Issue by scottransom** [nanograv/PINT#1005]
With JUMP'd TOAs in the .time file (ala old-school TEMPO method), the TOA reader sees that there are commands, but those are not turned into JUMPs in the timing model.

```
INFO: No jump flags to process from .tim file [pint.models.timing_model]
```

**Comment by scottransom** [comment]
After some investigation, this is a bug in the first `for` loop in `jump_flags_to_params()` in timing_model.py.  The else that triggers that INFO statement shouldn't be processed until each TOA has been checked for a "jump" key.  That probably also applies to the next `for` loop as well.

**Comment by scottransom** [comment]
Hmmm.  It is more than that.  The flags don't mention *which* JUMP they are for.  In my example, I have 4 different JUMP'd sections of TOAs (i.e. 4 different JUMPs), but they all have "jump" as a flag, and so you can't differentiate between them.  This probably needs to be changed in the TOA reader so that a different JUMP name (probably numbered) is assigned instead of the generic "jump".

**Comment by scottransom** [comment]
Note that that is already happening in toa.py.  The flag "jump" is set (which is a dictionary key), but the value for that dictionary is the JUMP number.  So we just need to properly use those JUMP numbers -- and in `jump_flags_to_params()` it looks like those numbers are trying to be used...  But they certainly don't seem to be part of the model!

**Comment by champagne-cmd** [comment]
Yes @aarchiba pointed out this error in one of my PRs, I plan on tackling this next. I commented this function out in the PR #968 in the meantime because I also need to update how it handles TOAs with multiple flags, and I figured it would be best to just fix that function up in its own PR.

**Comment by champagne-cmd** [comment]
@scottransom what .tim file(s) were you using? Are they a part of PINT, or can you send them to me/direct me to where I can download them?

**Comment by scottransom** [comment]
@champagne-cmd I emailed them to you.

**Comment by champagne-cmd** [comment]
Thank you @scottransom . When I run `get_TOAs` on that .tim file, it gives this error: `"Observatory name 'm' is not defined"`. How can I fix this?

**Comment by scottransom** [comment]
Right!  That is what my other PR #1004  is about.  If you merge that (it is really simple) in a dev branch, you can run things.

**Comment by champagne-cmd** [comment]
Ah I see, thank you!

**Comment by aarchiba** [comment]
What is the status of this? And is it affected by #1072?

**Comment by scottransom** [comment]
This bug, as reported is fixed.  But yes, I suspect that it is affected by #1072

**Comment by paulray** [comment]
Can be closed now? @scottransom

**Comment by scottransom** [comment]
Ah, yes!  Fixed by #1206
